lib/Runtime/PlatformAgnostic/Platform/Windows/UnicodeText.cpp

```diff
@@ -416,9 +416,19 @@ namespace PlatformAgnostic
 
         int LogicalStringCompare(const char16* string1, const char16* string2)
         {
-            // CompareStringW called with these flags is equivalent to calling StrCmpLogicalW
+            char16 localeName[LOCALE_NAME_MAX_LENGTH] = { 0 };
+
+            int ret = GetUserDefaultLocaleName(localeName, _countof(localeName));
+
+            if (ret == 0)
+            {
+                AssertMsg(FALSE, "GetUserDefaultLocaleName failed");
+                return -CSTR_EQUAL;
+            }
+
+            // CompareStringEx called with these flags is equivalent to calling StrCmpLogicalW
             // and we have the added advantage of not having to link with shlwapi.lib just for one function
-            int i = CompareStringW(LOCALE_USER_DEFAULT, NORM_IGNORECASE | SORT_DIGITSASNUMBERS, string1, -1, string2, -1);
+            int i = CompareStringEx(localeName, NORM_IGNORECASE | SORT_DIGITSASNUMBERS, string1, -1, string2, -1, NULL, NULL, 0);
```

Comment:
I believe you can pass either `""` or `nullptr` for the user default locale (not both -- one meant user default and the other meant invariant). That should make it so you don't have to call `GetUserDefaultLocaleName`.